jenkins-integration/jenkins-jobs/common/scripts/jenkins/pipeline.groovy

```diff
@@ -110,6 +110,19 @@ def step040_install_puppet_code(script_dir, code_deploy, server_era) {
     }
 }
 
+def step045_install_hiera_config(script_dir, code_deploy, server_era) {
+    hiera_config_source_file = (code_deploy["hiera_config_source_file"] == null) ?
+            "" : code_deploy["hiera_config_source_file"]
```

Comment:
> Kinda seems like we could check for nil here and just skip the rest of this step from here if it's nil, but I feel like I must be missing something about what you're trying to do.

I was allowing the environment variables to flow through as empty string if no corresponding keys for them were set in the `code_deploy` hash.  The code below this in the Ruby script layer would no-op copying in a hiera.yaml file and/or modifying the datadir in the hiera.yaml based on whether or not these environment variables have a non-empty value.  That's all these are here for.

> Also seems like you could check for the existence of the file here and if it doesn't exist, fail. I've probably been inconsistent about doing checks like that from the groovy code vs. the shell code or beaker scripts, but my gut is that it'd be better to have as much of that in the groovy as possible? WDYT?

I agree that checking for the existence of the file and failing would be a good thing to do.  It probably fails without those explicit checks anyway but not in a very understandable way.  I can add that in.

I think it might need to be done from the beaker code, though, since the files that I'd be checking for the existence of would be on the SUT and I don't think we have a straightforward way of accessing the SUT from where the Groovy code would be run, the Jenkins slave, right?

Also, I think I'd only want to check for the existence of `hiera_config_source_file` on the SUT and not the `hiera_config_datadir` since the `hiera_config_datadir` would often be an interpolated value and I don't really want any of this code to have to know anything about how to do the interpolation.  Sound reasonable?